WordPress/Classes/Services/UsersService.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+
+/// UserService is responsible for interacting with UserServiceRemoteXMLRPC to 
+/// fetch User and Profile related details from self-hosted blogs.  See the
+/// PeopleService for WordPress.com blogs via the REST API.
+///
+open class UsersService {
+
+    /// Fetch profile information for the user of the specified blog.
+    ///
+    func fetchProfile(blog: Blog, success: @escaping ((UserProfile) -> Void), failure: @escaping ((NSError?) -> Void)) {
+        guard let api = blog.xmlrpcApi, let username = blog.username, let password = blog.password else {
+            assertionFailure("Only self-hosted blogs are allowed.")
```

Comment:
Make sure to call the `failure` closure here otherwise the UI will forever wait.